jobmon_distributor/src/jobmon/distributor/distributor_instance.py

```diff
@@ -459,11 +461,12 @@ def refresh_status_from_db(self, status: str) -> None:
             except KeyError:
                 new_task_instance_ids.append(task_instance_id)
             else:
-                # remove from old status set
-                previous_status = task_instance.status
-                self._task_instance_status_map[previous_status].remove(
-                    task_instance
-                )
+                # Loop through all possible registers and try a discard. This prevents problems
+                # from task instances that have a mismatch between the in-memory state and the
+                # register they belong to. Performance shouldn't be a problem since there are
+                # only a set amount of active states.
+                for status, registry in self._task_instance_status_map.items():
```

Comment:
Comment: make a task to remove all task instance status mutations outside of refresh_status_from_db